scripts/install_third_party.py

```diff
@@ -165,20 +190,76 @@ def return_json(source_url):
     return json.loads(response)
 
 
+def test_manifest_syntax(dependency_type, dependency_dict):
+    """This checks syntax of the manifest.json dependencies.
+
+    Display warning message when there is an error and terminate the program.
+    Args:
+      dependency_type: dependency download format.
+      dependency_dict: manifest.json dependency dict
+    """
+    keys = dependency_dict.keys()
+    mandatory_keys = DOWNLOAD_FORMATS_TO_MANIFEST_KEYS[
+        dependency_type]['mandatory_keys']
+    optional_key_pairs= DOWNLOAD_FORMATS_TO_MANIFEST_KEYS[
+        dependency_type]['optional_key_pairs']
+    for key in mandatory_keys:
+        if key not in keys:
+            print '------------------------------------------'
+            print 'There is syntax error in this dependency'
+            print dependency_dict
+            print 'This key is missing or misspelled: "%s".' % key
+            print 'Exiting'
+            sys.exit()
+    if optional_key_pairs:
+        for optional_keys in optional_key_pairs:
+            if not any([True for key in optional_keys if key in keys]):
+                print '------------------------------------------'
+                print 'There is syntax error in this dependency'
+                print  dependency_dict
+                print 'This key is missing or misspelled: "%s".' % key
+                print 'Exiting'
+                sys.exit()
+    dependency_url = dependency_dict['url']
```

Comment:
Add newline above this one; add a comment saying something like

    # Checks the validity of the URL corresponding to the file format.